### PR TITLE
[BFN] syncd-rpc build with thrift 0.14.1

### PR DESCRIPTION
**Pull request by msosyak**
Depends on https://github.com/Azure/sonic-buildimage/pull/9881

#### Why I did it
The BFN SDK depends on thrift 0.14.1 and in the case syncd-rpc there is conflict in syncd as it should be linked with thrift 0.11.0(libsaithrift) and 0.14.1(libsai BFN impl) which are not binary compatible and caused runtime issues.

#### How I did it
- Added a new target thrift target https://github.com/Azure/sonic-buildimage/pull/9881 to build thrift 0.14.1
- Added patches to patch saithrfit sources to be compatible with thrift 0.14.1  (https://github.com/opencomputeproject/SAI/pull/1403)
- Added rules to build specific docker-ptf  for BFN platform
#### How to verify it
```
make init
NOJESSIE=1 NOSTRETCH=1 make configure PLATFORM=barefoot
NOJESSIE=1 NOSTRETCH=1 make  target/docker-syncd-bfn-rpc.gz
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: platform/barefoot/docker-ptf.mk [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# docker image for docker-ptf
```
[the remark this answers is not in the document]
Comment (msosyak) [reply platform/barefoot/docker-ptf.mk:1]: Yes, If we merge https://github.com/Azure/sonic-buildimage/pull/9881 and update https://github.com/Azure/sonic-buildimage/pull/9729 to use thrift 0.14.1 (a.k.a THRIFT_2) then there is no need in the separate docker-ptf image for Barefoot.

[the remark this answers is not in the document]
Comment (richardyu-ms) [reply platform/barefoot/docker-ptf.mk:1]: thank you, gotcha

**Review by richardyu-ms — DISMISSED** [review]

File: platform/barefoot/docker-ptf.mk (hunk 1 above)
[the remark this answers is not in the document]
Comment (richardyu-ms) [reply platform/barefoot/docker-ptf.mk:1]: so need to remove this file after those two PRs get merged?

[the remark this answers is not in the document]
Comment (msosyak) [reply platform/barefoot/docker-ptf.mk]: Already removed docker-ptf target from Barefoot platfrom

**Review by richardyu-ms — APPROVED** [review]

**Pull request merged**